runners/spark/src/main/java/org/apache/beam/runners/spark/translation/BoundedDataset.java

```diff
@@ -69,6 +70,14 @@
     return rdd;
   }
 
+  List<byte[]> getBytes(WindowedValue.WindowedValueCoder<T> wvCoder) {
+    if (clientBytes == null) {
+      JavaRDDLike<byte[], ?> bytesRDD = rdd.map(CoderHelpers.toByteFunction(wvCoder));
+      clientBytes = bytesRDD.collect();
```

Comment:
Is bytesRDD.collect expected to be invoked only once?

Reply:
`getBytes` implicitly assumes a 1:1 relationship between `this` and the `wvCoder` argument, meaning the value of `bytesRDD.collect` would always be the same for each BoundedDataset object and therefore would be cacheable. (This is similar for the existing `getValues` function.)

Reply:
I would ideally make these method stateless but I think its fine for now as its the same in getValue